code/ui/components/src/new/Select/Select.tsx

```diff
@@ -0,0 +1,180 @@
+  <SelectPrimitive.Portal>
+    <StyledContent ref={ref} {...props}>
+      <StyledScrollUpButton>
+        <Arrowup size={12} />
+      </StyledScrollUpButton>
+      <StyledViewport>{children}</StyledViewport>
+      <StyledScrollDownButton>
+        <Arrowdown size={12} />
+      </StyledScrollDownButton>
+    </StyledContent>
+  </SelectPrimitive.Portal>
+));
+SelectContent.displayName = SelectPrimitive.Content.displayName;
+
+const SelectLabel = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Label>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Label>
+>(({ className, ...props }, ref) => <SelectPrimitive.Label ref={ref} {...props} />);
+SelectLabel.displayName = SelectPrimitive.Label.displayName;
+
+const SelectItem = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Item>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Item>
+>(({ className, children, ...props }, ref) => (
+  <StyledItem ref={ref} {...props}>
+    <StyledItemIndicator>
+      <Check size={12} />
+    </StyledItemIndicator>
+    <SelectPrimitive.ItemText>{children}</SelectPrimitive.ItemText>
+  </StyledItem>
+));
+SelectItem.displayName = SelectPrimitive.Item.displayName;
+
+const SelectSeparator = React.forwardRef<
+  React.ElementRef<typeof SelectPrimitive.Separator>,
+  React.ComponentPropsWithoutRef<typeof SelectPrimitive.Separator>
+>(({ className, ...props }, ref) => <SelectPrimitive.Separator ref={ref} {...props} />);
+SelectSeparator.displayName = SelectPrimitive.Separator.displayName;
+
+export const Select = {
+  Root: SelectPrimitive.Root,
+  Group: SelectPrimitive.Group,
+  Value: SelectPrimitive.Value,
+  Trigger: SelectTrigger,
+  Content: SelectContent,
+  Label: SelectLabel,
+  Item: SelectItem,
+  Separator: SelectSeparator,
+};
+
+const StyledTrigger = styled(SelectPrimitive.Trigger)(({ theme }) => ({
+  all: 'unset',
+  display: 'flex',
+  width: '100%',
+  height: '32px',
+  alignItems: 'center',
+  justifyContent: 'space-between',
+  transition: 'box-shadow 200ms ease-out, opacity 200ms ease-out',
+  color: theme.input.color,
+  background: theme.input.background,
+  boxShadow: `${theme.input.border} 0 0 0 1px inset`,
+  borderRadius: theme.input.borderRadius,
+  fontSize: theme.typography.size.s2 - 1,
+  padding: '6px 10px',
+  boxSizing: 'border-box',
+  lineHeight: '20px',
+
+  '&:focus': {
+    boxShadow: `${theme.color.secondary} 0 0 0 1px inset`,
+    outline: 'none',
+  },
+
+  '&[disabled]': {
+    cursor: 'not-allowed',
+    opacity: 0.5,
+  },
+
+  '&[data-placeholder]': {
+    color: theme.textMutedColor,
+  },
+
+  '&:-webkit-autofill': {
+    WebkitBoxShadow: `0 0 0 3em ${theme.color.lightest} inset`,
+  },
+}));
+
+const StyledContent = styled(SelectPrimitive.Content)(({ theme }) => ({
+  boxSizing: 'border-box',
+  overflow: 'hidden',
+  backgroundColor: theme.input.background,
+  borderRadius: '6px',
+  border: theme.base === 'dark' ? `1px solid ${theme.input.border}` : 'none',
```

Comment:
Interesting choice to have no border at all in dark-mode. Can you elaborate?
Would a `1px solid transparent` not be better? Less layout shifts.